src/datatype/index.js

```diff
@@ -172,26 +177,73 @@ export class DataType extends TypedEmitter {
 
   /**
    * @param {string} docId
+   * @param {{ lang?: string }} [opts]
    */
-  async getByDocId(docId) {
+  async getByDocId(docId, { lang } = {}) {
     await this.#dataStore.indexer.idle()
-    const result = this.#sql.getByDocId.get({ docId })
+    const result = /** @type {MapeoDoc} */ (this.#sql.getByDocId.get({ docId }))
```

Comment:
Nitpick: this type should include `undefined`, because we don't know (until the next line) that the doc is found.

```suggestion
    const result = /** @type {undefined | MapeoDoc} */ (
      this.#sql.getByDocId.get({ docId })
    )
```